charts/geoserver/templates/statefulset.yaml

```diff
@@ -0,0 +1,95 @@
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: {{ include "geoserver.fullname" . }}
+  labels:
+{{ include "geoserver.labels" . | indent 4 }}
+spec:
+  replicas: {{ .Values.replicaCount }}
+  selector:
+    matchLabels:
+      app.kubernetes.io/name: {{ include "geoserver.name" . }}
+      app.kubernetes.io/instance: {{ .Release.Name }}
+  serviceName: {{ include "geoserver.fullname" . }}
+  template:
+    metadata:
+      labels:
+        app.kubernetes.io/name: {{ include "geoserver.name" . }}
+        app.kubernetes.io/instance: {{ .Release.Name }}
+    spec:
+    {{- with .Values.imagePullSecrets }}
+      imagePullSecrets:
+        {{- toYaml . | nindent 8 }}
+    {{- end }}
+      initContainers:
+        {{- if and .Values.storage.initDataUrl }}
+        - name: download-data
+          image: alpine
+          command: ["/bin/sh","/mnt/init-data.sh"]
+          volumeMounts:
+            - name: temp
+              mountPath: /mnt/download
+            - name: init-configmap
+              mountPath: /mnt/
+            - name: datadir
+              mountPath: {{ .Values.storage.datadir }}
+        {{- end }}
+      containers:
+        - name: {{ .Chart.Name }}
+          image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
+          env:
+            - name: STABLE_EXTENSIONS
+              value: {{ .Values.environment.stableExtensions }}
+            - name: EXTRA_JAVA_OPTS
+              value: {{ .Values.environment.javaOptions }}
+            {{- with .Values.extraEnv }}
+            {{- tpl . $ | nindent 12 }}
+            {{- end }}
+          envFrom:
+            {{- with .Values.extraEnvFrom }}
+            {{- tpl . $ | nindent 12 }}
+            {{- end }}
+          ports:
+            - name: http
+              containerPort: 8080
+              protocol: TCP
+          readinessProbe:
+            httpGet:
+              path: {{ .Values.probes.readinessUrl }}
+              port: http
+          livenessProbe:
+            httpGet:
+              path: {{ .Values.probes.livenessUrl }}
+              port: http
+              httpHeaders:
```

Comment:
Doesn't this depend on the configured livenessProbe URL?

Reply:
yes, good point. I'll remove it to also allow other request